simulators/optimism/daisy-chain/main.go

```diff
@@ -0,0 +1,266 @@
+			"HIVE_OP_GETH_SEQUENCER_HTTP":     GoerliSequencerRPC,
+			"HIVE_OP_GETH_HISTORICAL_RPC":     HistoricalSequencerRPC,
+		})
+		d.WaitUpOpL2Engine(0, time.Second*10)
+
+		// Seed the random number generator.
+		rand.Seed(time.Now().UnixNano())
+
+		optimism.RunTests(ctx, t, &optimism.RunTestsParams{
+			Devnet:      d,
+			Tests:       tests,
+			Concurrency: 40,
+		})
+	}
+}
+
+// Tests that a daisy-chained debug_traceBlockByNumber call to `op-geth` works as expected.
+func debugTraceBlockByNumberTest(t *hivesim.T, env *optimism.TestEnv) {
+	// Grab a random historical block
+	blockNumber := getRandomHistoricalBlockHex()
+
+	// Grab the result of the trace call from op-geth. This request should be daisy-chained.
+	opGeth := env.Devnet.GetOpL2Engine(0).RPC()
+	var opGethRes []*txTraceResult
+	err := opGeth.CallContext(env.Ctx(), &opGethRes, "debug_traceBlockByNumber", blockNumber)
+	require.NoError(t, err, "failed to call debug_traceBlockByNumber on op-geth")
+
+	// Grab the result of the trace call from the historical sequencer endpoint. The result
+	// should be the same.
+	histSeq, err := rpc.DialHTTP(HistoricalSequencerRPC)
+	require.NoError(t, err, "failed to dial historical sequencer RPC")
+	var histSeqRes []*txTraceResult
+	err = histSeq.CallContext(env.Ctx(), &histSeqRes, "debug_traceBlockByNumber", blockNumber)
+	require.NoError(t, err, "failed to call debug_traceBlockByNumber on historical sequencer")
+
+	// Compare the results.
+	require.Equal(t, histSeqRes, opGethRes, "results from historical sequencer and op-geth do not match")
+}
+
+// Tests that a daisy-chained debug_traceBlockByHash call to `op-geth` works as expected.
+func debugTraceBlockByHashTest(t *hivesim.T, env *optimism.TestEnv) {
+	// Grab a blockhash at a random block that is within the historical window.
+	block, err := env.Devnet.GetOpL2Engine(0).EthClient().BlockByNumber(env.Ctx(), getRandomHistoricalBlockNr())
+	require.NoError(t, err, "failed to get block by number")
+	blockHash := block.Hash()
+
+	// Grab the result of the trace call from op-geth. This request should be daisy-chained.
+	opGeth := env.Devnet.GetOpL2Engine(0).RPC()
+	var opGethRes []*txTraceResult
+	err = opGeth.CallContext(env.Ctx(), &opGethRes, "debug_traceBlockByHash", blockHash.String())
+	require.NoError(t, err, "failed to call debug_traceBlockByHash on op-geth")
+
+	// Grab the result of the trace call from the historical sequencer endpoint. The result
+	// should be the same.
+	histSeq, err := rpc.DialHTTP(HistoricalSequencerRPC)
+	require.NoError(t, err, "failed to dial historical sequencer RPC")
+	var histSeqRes []*txTraceResult
+	err = histSeq.CallContext(env.Ctx(), &histSeqRes, "debug_traceBlockByHash", blockHash.String())
+	require.NoError(t, err, "failed to call debug_traceBlockByHash on historical sequencer")
+
+	// Compare the results.
+	require.Equal(t, histSeqRes, opGethRes, "results from historical sequencer and op-geth do not match")
```

Comment:
This is basically the same code in a bunch of different tests just with a different method name and arg. Probably worth extracting it into a reusable method.